### PR TITLE
refactor(journal): posting journal account editing

### PR DESCRIPTION
**Pull request by jniles**
This PR implements posting journal account editing with a dropdown.  It should be helpful in finding and linking accounts correctly.

The editor is a dropdown based off the MIT-licensed [ui-grid-edit-datepicker](https://github.com/Joiler/ui-grid-edit-datepicker).  This helped me figure out how to properly hook up the events for onEdit and on $destroy, as well as detecting external clicks.

Partially addresses #1500.

---
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by mbayopanda** [comment]
@kwilu retry

**Comment by ghost** [comment]
@mbayopanda: :key: Insufficient privileges: Not in reviewers

**Comment by lomamech** [comment]
@kwilu try

**Comment by ghost** [comment]
@lomamech: :key: Insufficient privileges: Not in reviewers

**Comment by ghost** [comment]
:hourglass: Trying commit ad782aa31628119589a1bdc2cc4308f3a7585e71 with merge 7f9ecc01a3bf2814b5a91b3b9201ab68f2bb458e...

File: client/src/modules/journal/ui-grid-edit-account.directive.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+          $scope.stopEdit = function(evt) {
+            $scope.$emit(uiGridEditConstants.events.END_CELL_EDIT);
+          };
+
+          $scope.$on('$destroy', function () {
+            angular.element(window).off('click', onWindowClick);
+          });
+
+          $scope.setAccountOnRow = function (row, account) {
+            console.log('account:', account);
```
Comment (jniles) [inline client/src/modules/journal/ui-grid-edit-account.directive.js]: :-1:

**Comment by ghost** [comment]
:umbrella: The latest upstream changes (presumably dd072023ec6ec4fe82c71479d273394d2d698f67) made this pull request unmergeable. Please resolve the merge conflicts.

**Comment by jniles** [comment]
@kwilu try

**Comment by ghost** [comment]
:hourglass: Trying commit 30ef44e896b7d064ff7e7d751f623ca9a16ee09d with merge 5927be5a9671dad68692887ce599e031771e99be...

**Comment by jniles** [comment]
@kwilu retry

**Comment by ghost** [comment]
:hourglass: Trying commit 30ef44e896b7d064ff7e7d751f623ca9a16ee09d with merge 868112da20cd50cfd59c0de7fc92ad4fb400f0f3...

**Pull request merged**